packages/react/src/primitives/utils/useLayoutEffect.ts

```diff
@@ -0,0 +1,15 @@
+// Source: https://github.com/radix-ui/primitives/blob/7ae63b6cce6ea53ea5d65b6d411894c004b38f47/packages/react/use-layout-effect/src/useLayoutEffect.tsx
+import * as React from 'react';
+
+/**
+ * On the server, React emits a warning when calling `useLayoutEffect`.
+ * This is because neither `useLayoutEffect` nor `useEffect` run on the server.
+ * We use this safe version which suppresses the warning by replacing it with a noop on the server.
+ *
+ * See: https://reactjs.org/docs/hooks-reference.html#uselayouteffect
+ */
+const useLayoutEffect = Boolean(globalThis?.document)
```

Comment:
For SSR, we should call `useEffect` instead of a `noop` function. This is what some other popular libraries do:

https://github.com/reduxjs/react-redux/blob/d16262582b2eeb62c05313fca3eb59dc0b395955/src/components/connectAdvanced.js#L40
https://github.com/atlassian/react-beautiful-dnd/blob/master/src/view/use-isomorphic-layout-effect.js

Reply:
I've seen that before too, but recently I've been wondering if it's a valid pattern. [useEffect is a noop on the server](https://github.com/reduxjs/react-redux/blob/d16262582b2eeb62c05313fca3eb59dc0b395955/src/components/connectAdvanced.js#L36) anyway, so why not be more declarative with a noop function instead? I've also seen warnings about `useEffect` when it's called on the server side, so this reducing unnecessary noise.

Reply:
cool, i'm fine with keeping it as noop